go.mod

```diff
@@ -48,17 +48,17 @@ require (
 	github.com/pbnjay/memory v0.0.0-20210728143218-7b4eea64cf58
 	github.com/prometheus/client_golang v1.14.0
 	github.com/prometheus/client_model v0.4.0
-	github.com/quic-go/quic-go v0.37.5
+	github.com/quic-go/quic-go v0.37.2-0.20230818062947-f45847be65fc
```

Comment:
https://github.com/libp2p/go-libp2p/pull/2506 updates quic-go to a version that now includes the API change. Let's rebase this PR once it's merged.